### ISSUE TITLE
Dropdown controls and its list items are not accessible by label name while we are using voice access.

### ISSUE THREAD

**Issue by msftedad** [apache/superset#24857]
Dropdown controls and its list items are not accessible by label name while we are using voice access.

#### How to reproduce the bug
1. Open URL: https://titanweb-testenv/superset/dashboard/268/ in Edge browser.
2. Navigate to the Market filter section appearing.
3. Turn ON Voice Access. 
4. Say 'click Type or Select Market'.
5. Observe that dropdown controls and its list items are not accessible by label name while we are using voice access.

### Expected results

Dropdown controls should be accessible by label name while we are using voice access.

### Actual results

Dropdown controls and its list items are not accessible by label name while we are using voice access.

### Environment

OS: Windows_11 
Version: 22H2
OS Build: 22610.1
Browser: New edge (Version 105.0.1343.53)
URL: https://titanweb-testenv/superset/dashboard/268/
Tool: [Voice Access Wiki](https://nam06.safelinks.protection.outlook.com/?url=https:%2f%2fwww.osgwiki.com%2fwiki%2fVoiceAccess_Testing&data=05%7c01%7cv-palvigupta%40microsoft.com%7cf0ac64527dba479f048908da7b6662c2%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637957976929859524%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=yLl5z40XNgbHmD3usjPwHXOOoZIW5H2r7nF6E%2fq%2BUBg%3D&reserved=0)

### Additional context
Same issue repro on all dropdown controls.

**Attachments:**
![Dropdown controls and its list items are not accessible by label name while we are using voice access](https://github.com/apache/superset/assets/93735775/6854db3a-d02c-47c9-beaa-a985f73fd383)

https://github.com/apache/superset/assets/93735775/b8e73a58-8ae5-436d-85ac-792299ed8622

**Comment by rusackas** [comment]
Hi there,

I've noticed you filed a wave of these accessibility shortcomings. We could probably open a hundred more of these, but that just goes to underscore the point that Superset is not really up to snuff in this department. Would you have any interest in contributing to Superset in this realm and/or coordinating a community effort in this area?

**Comment by rusackas** [comment]
It's been a long while since this was opened. I've assigned it to a couple folks who might be able to get it onto an acessibility project board, but unless someone wants to contribute toward this in the near future, I think we'll likely close it since it's gone silent for so long, and it's more feature request than bug (though I suppose it depends who you ask!). We'd really love to see more contributions in the accessibility realm if anyone here wants to take a crack at it.

**Comment by rusackas** [comment]
Anyone object to closing this?

**Comment by rusackas** [comment]
Closing as `not planned`.
